coderd/templates.go

```diff
@@ -855,6 +861,35 @@ func (api *API) patchTemplateMeta(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, api.convertTemplate(updated))
 }
 
+func (api *API) notifyUsersOfTemplateDeprecation(ctx context.Context, template database.Template) error {
+	users, err := api.Database.GetUsersWithAccessToTemplateByID(ctx, template.ID)
```

Comment:
Also: double-check that the correct actor is available in the ctx is being used here for dbauthz. We may need to use `AsSystemRestricted()` for now, but I'd like to phase that out in favour of a separate system-internal RBAC role.

Reply:
EDIT: from reading the template admin and org template admin roles, it seems like that should be fine